frontend/src/containers/Collection/CollectionManage/CollectionManage.js

```diff
@@ -0,0 +1,167 @@
+
+import "./CollectionManage.css";
+
+class CollectionManage extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            collectionName: "",
+            collectionDescription: "",
+        };
+    }
+
+    componentDidMount() {
+        const collectionId = this.props.location.pathname.split("=")[1].split("/")[0];
+        this.props.onGetCollection({ id: collectionId })
+            .then(() => {
+                if (this.props.collectionStatus !== collectionStatus.SUCCESS) {
+                    this.props.history.push("/main");
+                } else {
+                    this.setState({
+                        collectionName: JSON.parse(JSON.stringify(
+                            this.props.selectedCollection.title,
+                        )),
+                        collectionDescription: JSON.parse(JSON.stringify(
+                            this.props.selectedCollection.text,
+                        )),
+                    });
+                }
+                return null;
+            });
+    }
+
+    updateCollectionHandler = () => {
+        this.props.onUpdateCollectionInfo({
+            id: this.props.selectedCollection.id,
+            title: this.state.collectionName,
+            text: this.state.collectionDescription,
+        });
+        // message or popup that says "collection is updated" may need to be implemented
+    }
+
+    render() {
+        return (
+            <div className="CollectionManage">
+                <div className="EditCollectionInfo">
+                    <div className="EditCollectionName">
+                        <h3 id="editNameTag">Collection Name</h3>
+                        <textarea
+                          id="editName"
+                          rows="1"
+                          type="text"
+                          value={this.state.collectionName}
+                          onChange={(event) => this.setState({
+                              collectionName: event.target.value,
+                          })}
+                        />
+                    </div>
+                    <div className="EditCollectionDesc">
+                        <h3 id="editDescTag">Collection Description</h3>
+                        <textarea
+                          id="editDescription"
+                          rows="3"
+                          type="text"
+                          value={this.state.collectionDescription}
+                          onChange={(event) => this.setState({
+                              collectionDescription: event.target.value,
+                          })}
+                        />
+                    </div>
+                    <div className="EditButtons">
+                        <Button
+                          id="UpdateCollectionButton"
+                          onClick={this.updateCollectionHandler}
+                          disabled={this.state.collectionName === ""}
```

Comment:
Suggestion: How about disabled also when there are no changes, as in AccountSetting? If so, we can inform users that the update is completed by disabling the button.(Because the changed state will have become the default state at the time!)

Reply:
OK, I reflected your suggestion.